### PR TITLE
[Pool Simulator] add Depth cost for Gyro 3CLP

### PR DESCRIPTION
**Pull request by luizakp**
This PR adds:
- Gyro 3CLP Depth cost chart
- add _checkIfInIsOnLimit to Gyro3

### CODE REVIEW

File: apps/balancer-tools/src/app/poolsimulator/(components)/DepthCost.tsx [hunk 1]
```diff
@@ -315,3 +308,25 @@ export function calculateDepthCost(
       return { amount: 0, type: "" };
   }
 }
+
+function getAlphaAndBeta(poolType: PoolTypeEnum, data: AnalysisData) {
```
Comment (devjoaov) [inline apps/balancer-tools/src/app/poolsimulator/(components)/DepthCost.tsx:312]: I would rename to something like `computeAlphaBetaValues`

**Review by devjoaov — DISMISSED** [review]
LGTM!

File: packages/math-poolsimulator/src/gyroE/index.test.ts [hunk 2]
```diff
@@ -10,76 +10,91 @@ import { ExtendedGyroEV2 } from "./index";
 
 describe("Tests new Gyro ECLP math function based on package other functions", () => {
   const poolData = poolsFromFile["pool"];
-  const percentages = [25, 50, 75, 95];
+  const percentages = [25, 50, 75, 80];
```
Comment (luizakp) [inline packages/math-poolsimulator/src/gyroE/index.test.ts:13]: @yvesfracari I don't know why, but the highest percentage to converge for GyroE was 80 🤔

Comment (yvesfracari) [reply packages/math-poolsimulator/src/gyroE/index.test.ts:13]: Hummm! That is strange, looking at it rn

File: apps/balancer-tools/src/lib/schema.ts [hunk 3]
```diff
@@ -253,7 +253,7 @@ export const FxSchema = BasePoolSchema.extend({
 // TODO: BAL-518 Rename this to StableSwapSchema when pool simulator is finished
 export const StableSwapSimulatorDataSchema = BasePoolSchema.extend({
   ampFactor: z.coerce.number().positive().min(1).max(5000), //source: https://github.com/balancer/balancer-v2-monorepo/blob/c4cc3d466eaa3c1e5fa62d303208c6c4a10db48a/pkg/pool-stable/contracts/StableMath.sol#L28
-  tokens: z.array(BaseTokenSchema).length(2).min(2).max(5),
+  tokens: z.array(BaseTokenSchema).min(2).max(5),
```
Comment (yvesfracari) [inline apps/balancer-tools/src/lib/schema.ts:256]: 👍

File: packages/math-poolsimulator/src/gyro3/index.test.ts [hunk 4]
```diff
@@ -11,57 +11,74 @@ import { ExtendedGyro3 } from "./index";
...
-          swapFee: poolData.swapFee,
-          totalShares: poolData.totalShares,
-          tokens: poolData.tokens,
-          tokensList: poolData.tokensList,
-          root3Alpha: poolData.root3Alpha,
-        }),
-      );
+      percentages.forEach((percentage) => {
+        describe(`Tests for ${percentage}% with ${tokenIn} as tokenIn and ${tokenOut} as tokenOut`, () => {
+          const amm = new AMM(
```
Comment (yvesfracari) [inline packages/math-poolsimulator/src/gyro3/index.test.ts:22]: is this diff right?

**Review by yvesfracari — CHANGES_REQUESTED** [review]
The Gyro E 95 percent test that didn't run actually is a problem with the algorithm. Please take a look at my PR before merging this.

File: packages/math-poolsimulator/src/gyro3/index.test.ts (hunk 4 above)
Comment (luizakp) [reply packages/math-poolsimulator/src/gyro3/index.test.ts:22]: yes I just moved the 

```
percentages.forEach((percentage) => {
    describe(`Tests for ${percentage}% with ${tokenIn} as tokenIn`, () => {
      const amm = new AMM(
        new ExtendedGyro3({
          swapFee: poolData.swapFee,
          totalShares: poolData.totalShares,
          tokens: poolData.tokens,
          tokensList: poolData.tokensList,
          root3Alpha: poolData.root3Alpha,
        }),
      );
```

inside and git made it weird kkkk

**Review by yvesfracari — APPROVED** [review]

**Pull request merged**